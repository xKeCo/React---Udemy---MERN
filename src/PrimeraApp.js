import React from "react";
import PropTypes from "prop-types";

const PrimeraApp = ({ saludo, profesor, plataforma }) => {
  return (
    <>
      <h2>{saludo}</h2>
      <h3>
        I'm improving my skills on React.js in {plataforma} with {profesor}
      </h3>
    </>
  );
};

PrimeraApp.propTypes = {
  profesor: PropTypes.string.isRequired,
  plataforma: PropTypes.string.isRequired,
};

PrimeraApp.defaultProps = {
  profesor: "Fernando Herrrera",
  plataforma: "Udemy.com",
};

export default PrimeraApp;
