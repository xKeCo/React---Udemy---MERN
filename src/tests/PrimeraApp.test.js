import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import PrimeraApp from "../PrimeraApp";
// import { render } from "@testing-library/react";

describe("Pruebas en el componente de PrimeraApp", () => {
  // test('Debe mostrar el mensaje "Hi,I\'m Kevin"', () => {
  //   const saludo = "Hi,I'm Kevin";
  //   const { getByText } = render(<PrimeraApp saludo="Hi,I'm Kevin" />);
  //   expect(getByText(saludo)).toBeInTheDocument();
  // });

  test("debe de mostrar PrimeraApp correctamente", () => {
    const saludo = "Hi,I'm Kevin";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar la plataforma enviada por props", () => {
    const saludo = "Hi,I'm Kevin";
    const plataforma = "Udemy.com";
    const profesor = "Fernando Herrrera";

    const wrapper = shallow(
      <PrimeraApp saludo={saludo} plataforma={plataforma} profesor={profesor} />
    );

    const textoParrafo = wrapper.find("h3").text();

    expect(textoParrafo).toBe(
      `I'm improving my skills on React.js in ${plataforma} with ${profesor}`
    );
  });
});
