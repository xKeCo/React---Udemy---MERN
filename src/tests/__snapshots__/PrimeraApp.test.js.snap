// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente de PrimeraApp debe de mostrar PrimeraApp correctamente 1`] = `
<Fragment>
  <h2>
    Hi,I'm Kevin
  </h2>
  <h3>
    I'm improving my skills on React.js in 
    Udemy.com
     with 
    Fernando Herrrera
  </h3>
</Fragment>
`;
